File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        sset = set()
        s = set(nums)
        if len(s) ==1 :
            return 1
        for i in nums:
            if i+1 in s and i-1 not in s:
                sset.add(i)

        cur = 0
        res = 0
        
        for i in sset:
            print(i)
            cur = 0
            while i in s:
                s.remove(i)
                i+=1
                cur+=1
            res = max(cur,res)
        return res

        

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-3.java
class Solution {
    public int characterReplacement(String s, int k) {
        String p = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        int res = 0;
        for(char c : p.toCharArray()){
            int l = 0;
            int r = 0;
            int cur = 0;
            int last = -1;
            
            while(r < s.length()){
                if(s.charAt(r) == c){}
                else{
                    if(cur < k){cur+=1;}
                    else{
                        //res = Math.max(res,r-l);
                        boolean f = false;
                        while(!f){if(s.charAt(l) != c){f = true;};l+=1;}
                        //cur-=1;
                        
                    }
                }
                res = Math.max(res,r-l+1);
                r+=1;
                
            }
            
        }
        return res;
    }
}
